staging/src/k8s.io/apiserver/pkg/storage/etcd3/watcher.go

```diff
@@ -402,19 +402,20 @@ func (wc *watchChan) prepareObjs(e *event) (curObj runtime.Object, oldObj runtim
 		// This is done here since we want to set the logical cluster the object is part of,
 		// without storing the clusterName inside the etcd object itself (as it has been until now).
 		// The etcd key is ultimately the only thing that links us to a cluster
-		if clusterName != "" {
-			if s, ok := curObj.(metav1.ObjectMetaAccessor); ok {
-				klog.Infof("SUB: %s", clusterName)
-				s.GetObjectMeta().SetClusterName(clusterName)
-			} else if s, ok := curObj.(metav1.Object); ok {
-				klog.Infof("SUB: %s", clusterName)
-				s.SetClusterName(clusterName)
-			} else if s, ok := curObj.(*unstructured.Unstructured); ok {
-				klog.Infof("SUB: %s", clusterName)
-				s.SetClusterName(clusterName)
-			} else {
-				klog.Infof("NO SUB: %T %s", curObj, clusterName)
-			}
+		if clusterName == "" {
```

Comment:
You can't make this change because it would break normal usage of kubernetes starting kube-apiserver.  That's what can't change in the serving chain.

Principle: our "added hacks" still have to not break the behavior of the k/k repo.

Reply:
I agree on the principle of not to break the behavior of the k/k repo so I obviously missed some of the impacts here.

Reply:
Do you mean that I should only emit error logs but **not fail** at all the places where I check for valid cluster presence in the `storage/etcd3` files such as the follownig places:
- https://github.com/kcp-dev/kubernetes/pull/7/files#diff-83db812e87640d3b07e697ed074e5e7a8f349429abf033fbb34e98dd8b82dac5R154
- https://github.com/kcp-dev/kubernetes/pull/7/files#diff-83db812e87640d3b07e697ed074e5e7a8f349429abf033fbb34e98dd8b82dac5R673
- etc ?

Reply:
Can we add some flag to reject requests without a cluster context? I agree we shouldn't break k/k but it's also useful in our usage to reject requests that don't make sense for us.